dex/src/main/java/io/crate/breaker/RamAccounting.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.breaker;
+
+/**
+ * Accounts for RAM usage by counting allocated bytes.
+ * Throws a CircuitBreakingException to avoid OutOfMemoryErrors,
+ * in case it thinks that too many bytes have been allocated.
+ */
+public interface RamAccounting {
+
+    /**
+     * Accounts for the supplied number of bytes
+     * @throws RuntimeException if too many bytes have been added.
+     */
+    void addBytes(long bytes) throws RuntimeException;
```

Comment:
RuntimeExceptions are usually not declared via `throws` clause but only as comment because they're usually unchecked.
I think it would also make sense to update the `@throws` comment to state that it throws a `CircuitBreakingException` to match the comment on the top-level.

Reply:
I couldn't add the `CircuitBreakingException` here because moving it to `dex` proved to be a major refactoring. Will update the comment.